core/domain/state_domain.py

```diff
@@ -229,6 +242,10 @@ def validate(
                     'to be <skill_id>-<misconception_id>, received %s'
                     % self.tagged_skill_misconception_id)
 
+        if len(self.rule_specs) == 0 and validation_from_exploration:
+            raise utils.ValidationError(
+                'There must be at least one rule for each answer group.')
```

Comment:
Isn't this true for `question`'s state as well?

Reply:
We have validation for the `question` as well just below it. So in case we have `training data` available and no rule specs then it will be okay but that will not be the case in `Exploration`